core/src/main/java/ai/timefold/solver/core/impl/domain/variable/declarative/GraphStructure.java

```diff
@@ -0,0 +1,87 @@
+package ai.timefold.solver.core.impl.domain.variable.declarative;
+
+import java.util.Arrays;
+
+import ai.timefold.solver.core.impl.domain.solution.descriptor.SolutionDescriptor;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public enum GraphStructure {
+    /**
+     * A graph structure that only accepts the empty graph.
+     */
+    EMPTY,
+
+    /**
+     * A graph structure without dynamic edges.
+     */
+    NO_DYNAMIC_EDGES,
+
+    /**
+     * A graph structure where there is at most
+     * one directional parent for each graph node.
```

Comment:
Deserves more explanation. Why is the one directional parent important?
Give examples of where it helps.